net/Socket.cpp

```diff
@@ -1553,7 +1553,7 @@ bool StreamSocket::checkRemoval(std::chrono::steady_clock::time_point now)
                                                     << ", " << *this);
         if (_socketHandler)
         {
-            _socketHandler->onDisconnect();
+            ensureDisconnected();
```

Comment:
I suppose now we can move this outside the `if (_socketHandler)` and have a single
```
            shutdown(); // signal
            shutdownConnection(); // real -> setShutdown()
```